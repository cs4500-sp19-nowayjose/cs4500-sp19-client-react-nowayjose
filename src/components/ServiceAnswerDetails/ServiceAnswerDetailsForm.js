import React from 'react'

export default function ServiceAnswerDetailsForm({
  serviceAnswers,
  serviceAnswer,
  selectServiceAnswer
}) {
  return(
      <div>
          <h3>Service Answer Details</h3>
          <select
              value={serviceAnswer.id}
              onChange={(e) => selectServiceAnswer(e.target.value)}
              className="answer-details-select-id form-control">
              {
                  serviceAnswers
                      .map(serviceAnswer =>
                          <option
                              className="select-other-answer"
                              value={serviceAnswer.id}
                              key={serviceAnswer.id}>
                              {serviceAnswer.choiceAnswer}
                          </option>
                      )
              }
          </select>
          <label>Service Answer Answer</label><br/>
          <input
              className="form-control answers-choice-answer"
              onChange={() => {}}
              value={"" + serviceAnswer.choiceAnswer}/>
          <input
            className="form-control answers-max-answer"
            value={"" + serviceAnswer.maxRangeAnswer} />
          <input
            className="form-control answers-min-answer"
            value={"" + serviceAnswer.minRangeAnswer} />
      </div>
  )
}
