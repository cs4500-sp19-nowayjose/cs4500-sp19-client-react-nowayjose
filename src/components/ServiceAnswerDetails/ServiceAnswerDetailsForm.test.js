import React from 'react'
import ServiceAnswerDetailsForm from './ServiceAnswerDetailsForm'
import renderer from 'react-test-renderer'
import serviceAnswers from '../../data/serviceAnswers.mock.json'
import {BrowserRouter as Router} from 'react-router-dom'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
      <ServiceAnswerDetailsForm
        serviceAnswers={serviceAnswers}
        serviceAnswer={serviceAnswers[0]}
        selectServiceAnswer={(id) => {}}
        />
      </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
