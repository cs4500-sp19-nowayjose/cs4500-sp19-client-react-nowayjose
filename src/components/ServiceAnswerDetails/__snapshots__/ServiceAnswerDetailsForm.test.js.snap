// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Service Answer Details
  </h3>
  <select
    className="form-control"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    />
    <option
      value={2}
    >
      0
    </option>
  </select>
  <label>
    Service Answer Answer
  </label>
  <br />
  <input
    className="form-control"
    onChange={[Function]}
    value={null}
  />
</div>
`;
